File: Gasto/src/main/java/com/py/api/model/dto/JwtResponse.java
package com.py.api.model.dto;

import com.py.api.model.sistema.Empresa;
import com.py.api.model.sistema.Licencia;
import com.py.api.model.sistema.Usuario;

import lombok.AllArgsConstructor;
import lombok.Data;

@Data
@AllArgsConstructor
public class JwtResponse {

	private final String jwttoken;
	private final Usuario usuario;
	private final Licencia licencia;
	private final Empresa empresa;

}

File: app_gasto/lib/src/module/login/models/jwt_response.dart
import 'package:app_gasto/src/module/login/models/empresa.dart';
import 'package:app_gasto/src/module/login/models/licencia.dart';
import 'package:app_gasto/src/module/login/models/usuario.dart';
import 'package:json_annotation/json_annotation.dart';

part 'jwt_response.g.dart';

@JsonSerializable()
class JwtResponse {
  String? jwttoken;
  Usuario usuario;
  Empresa? empresa;
  Licencia? licencia;

  JwtResponse({
    required this.jwttoken,
    required this.usuario,
    this.empresa,
    this.licencia,
  });

  factory JwtResponse.fromJson(Map<String, dynamic> json) =>
      _$JwtResponseFromJson(json);
  Map<String, dynamic> toJson() => _$JwtResponseToJson(this);
}
